adapters/improvedigital/improvedigital.go

```diff
@@ -144,3 +153,68 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 		Message: fmt.Sprintf("Failed to find impression for ID: \"%s\"", impID),
 	}
 }
+
+// This method responsible to clone request and convert additional consent providers string to array when additional consent provider found
+func (a *ImprovedigitalAdapter) getAdditionalConsentProvidersUserExt(request openrtb2.BidRequest) []byte {
+	const (
+		consentProvidersSettingsInputKey = "ConsentedProvidersSettings"
+		consentProvidersSettingsOutKey   = "consented_providers_settings"
+		consentedProvidersKey            = "consented_providers"
+	)
+
+	var (
+		err   error
+		cpStr string
+	)
+
+	// If user/user.ext not defined, no need to parse additional consent
+	if request.User == nil || request.User.Ext == nil {
+		return []byte{}
+	}
+
+	// Start validating additional consent
+	// Check key exist user.ext.ConsentedProvidersSettings
+	var userExtMap = make(map[string]json.RawMessage)
+	if err = json.Unmarshal(request.User.Ext, &userExtMap); err != nil {
+		return []byte{}
+	}
+
+	cpsMapValue, cpsJSONFound := userExtMap[consentProvidersSettingsInputKey]
+	if !cpsJSONFound {
+		return []byte{}
+	}
+
+	// Check key exist user.ext.ConsentedProvidersSettings.consented_providers
+	var cpMap = make(map[string]json.RawMessage)
+	if err = json.Unmarshal(cpsMapValue, &cpMap); err != nil {
+		return []byte{}
+	}
+
+	cpMapValue, cpJSONFound := cpMap[consentedProvidersKey]
+	if !cpJSONFound {
+		return []byte{}
+	}
+	// End validating additional consent
+
+	// Check if string contain ~, then substring after ~ to end of string
+	consentStr := string(cpMapValue)
+	var tildaPosition int
+	if tildaPosition = strings.Index(consentStr, "~"); tildaPosition == -1 {
+		return []byte{}
+	}
+	cpStr = consentStr[tildaPosition+1 : len(consentStr)-1]
+
+	// Prepare consent providers string
+	cpStr = fmt.Sprintf("[%s]", strings.Replace(cpStr, ".", ",", -1))
+	cpMap[consentedProvidersKey] = json.RawMessage(cpStr)
+
+	cpJSON, _ := json.Marshal(cpMap)
```

Comment:
Please do not ignore errors. May you please return nil instead of []byte and error if any: 
```
if err != nil {
    return nil, err
}
```
Upper level expression `len(userExtAddtlConsent) > 0` handled nil properly.

Reply:
Yeah i second this